webapp/src/components/boardsSwitcherDialog/boardSwitcherDialog.tsx

```diff
@@ -81,6 +85,7 @@ const BoardSwitcherDialog = (props: Props): JSX.Element => {
             title={title}
             subTitle={subTitle}
             searchHandler={searchHandler}
+            refs={refs}
```

Comment:
All this refs passings looks a bit convoluted to me. Couldn't it be something like an state at this level with the selected id? and an onSelect callback or something like that?

I always find passing refs down the components really hard to track.

Reply:
I agree, it wasn't as straightforward as I was expecting it to be. My reasoning for using `ref`s was it let me use the native `.focus()` and `.click()` methods on the underlying native elements.

Since the parent component (this) defines the search handler which returns the JSX components and passes that definition into the child component which calls it and does the rendering, it seemed to make sense to define the `ref`s in the parent so the child could use the native methods I mentioned above. `onSelect` did come to mind, but it's not available for all elements, only `<input>` and `<textarea>`.

Reply:
Ok, what I meant is. move the `selected` state to the `BoardSwitcherDialog` component, update the `selected` state using a callback that is passed to the SearchDialog component, and then, you can use a `useEffect` call to update the focus on every `selected` change. The syntethic click, is not going to be needed, because you are already there, and you have access to the `selectBoard` function. You are going to still need the refs, but you are not passing them around that is what makes it more confusing.